engines/ospray/ispc/geometry/SDFBeziers.ispc

```diff
@@ -0,0 +1,444 @@
+        {
+            uniform float q = bezier(t, bc->p0.x, bc->c0.x, bc->p1.x, bc->c1.x);
+            mi.x = min(mi.x, q);
+            ma.x = max(ma.x, q);
+        }
+
+        t = (-b.x + h.x) / a.x;
+        if (!isnan(t) && t > 0.0f && t < 1.0f)
+        {
+            uniform float q = bezier(t, bc->p0.x, bc->c0.x, bc->p1.x, bc->c1.x);
+            mi.x = min(mi.x, q);
+            ma.x = max(ma.x, q);
+        }
+    }
+
+    if (h.y > 0.0f)
+    {
+        h.y = sqrt(h.y);
+        uniform float t = (-b.y - h.y) / a.y;
+        if (!isnan(t) && t > 0.0f && t < 1.0f)
+        {
+            uniform float q = bezier(t, bc->p0.y, bc->c0.y, bc->p1.y, bc->c1.y);
+            mi.y = min(mi.y, q);
+            ma.y = max(ma.y, q);
+        }
+        t = (-b.y + h.y) / a.y;
+        if (!isnan(t) && t > 0.0f && t < 1.0f)
+        {
+            uniform float q = bezier(t, bc->p0.y, bc->c0.y, bc->p1.y, bc->c1.y);
+            mi.y = min(mi.y, q);
+            ma.y = max(ma.y, q);
+        }
+    }
+
+    if (h.z > 0.0f)
+    {
+        h.z = sqrt(h.z);
+        uniform float t = (-b.z - h.z) / a.z;
+        if (!isnan(t) && t > 0.0f && t < 1.0f)
+        {
+            uniform float q = bezier(t, bc->p0.z, bc->c0.z, bc->p1.z, bc->c1.z);
+            mi.z = min(mi.z, q);
+            ma.z = max(ma.z, q);
+        }
+        t = (-b.z + h.z) / a.z;
+        if (!isnan(t) && t > 0.0f && t < 1.0f)
+        {
+            uniform float q = bezier(t, bc->p0.z, bc->c0.z, bc->p1.z, bc->c1.z);
+            mi.z = min(mi.z, q);
+            ma.z = max(ma.z, q);
+        }
+    }
+
+    return make_box3fa(mi - r, ma + r);
+#endif
+}
+//////////////////////////////////////////////////////////////////////
+
+#define MIN_SEGMENT_COUNT 3
+#define MAX_SEGMENT_COUNT 9
+
+int screenAwareSegmentCount(const Bezier_ptr bc, const vec4f& params)
+{
+    const vec3f eye = make_vec3f(params);
+    const float segmentCountMultiplier = params.w;
+
+    float costhetaEye = dot(normalize(bc->p0 - eye), normalize(bc->p1 - eye));
+
+    float ratio = (1.0f - costhetaEye) / 0.5f; // cos(vfov / 180 * M_PI);
+
+    // TODO twick this
+
+    return min(max((int)(MIN_SEGMENT_COUNT +
+                         (10 * sqrt(ratio)) *
+                             (MAX_SEGMENT_COUNT - MIN_SEGMENT_COUNT)),
+                   MIN_SEGMENT_COUNT),
+               MAX_SEGMENT_COUNT) *
+           segmentCountMultiplier;
+}
+
+#undef MIN_SEGMENT_COUNT
+#undef MAX_SEGMENT_COUNT
+
+//////////////////////////////////////////////////////////////////////
+
+float bezierDistance(const vec3f& p, dataPtr_t geo_, dataPtr_t bezier_,
+                     const vec4f& params)
+{
+    const Geo_ptr geo = (const Geo_ptr)geo_;
+    const Bezier_ptr bc = (const Bezier_ptr)bezier_;
+
+    // TODO should be based on pixel footprint
+    const int segmentCount = screenAwareSegmentCount(bc, params);
+
+    const float deltaT = 1.0f / segmentCount;
+
+    float minDist = 1e10f;
```

Comment:
```suggestion
    float minDist = SDF_EPSILON;
```

Reply:
Nope in this case I should use `INFINITY`

Reply:
Ah, yes :)